plugins/woocommerce/src/Admin/Features/Settings.php

```diff
@@ -84,7 +84,7 @@ public static function add_component_settings( $settings ) {
 		$setting_pages = \WC_Admin_Settings::get_settings_pages();
 		$pages         = array();
 		foreach ( $setting_pages as $setting_page ) {
-			$pages = $setting_page->add_settings_page( $pages );
+			$pages = $setting_page->get_settings_page_data( $pages );
```

Comment:
It looks like we're loading `window.wcSettings?.admin?.settingsPages` for every admin pages but we only need it for WooCommerce settings pages. I think this is a potential performance improvement. We could consider adding an API to get the settings pages data. The downside is that there will need a loading spinner when the user navigates to the WooCommerce settings pages but since the settings page data could be very large, it might be worth it. Just a thought and no need to handle this in this PR.

Reply:
We could also check for the page I.D and conditionally set `window.wcSettings?.admin?.settingsPages`. Example: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/Features/LaunchYourStore.php#L108

Reply:
Good call team, in https://github.com/woocommerce/woocommerce/pull/52472/commits/53518e2209d56a4e9242fc7168c14dcd2145f898 I added a helper method `is_settings_page` and use it before continuing with gathering the data.